web/src/components/core/AlertOutOfSync.tsx

```diff
@@ -0,0 +1,115 @@
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import {
+  Alert,
+  AlertActionCloseButton,
+  AlertGroup,
+  AlertProps,
+  Content,
+} from "@patternfly/react-core";
+import Link from "./Link";
+import Text from "./Text";
+import { useInstallerClient } from "~/context/installer";
+import { isEmpty } from "radashi";
+import { _ } from "~/i18n";
+
+type AlertOutOfSyncProps = Partial<AlertProps> & {
+  /**
+   * The scope to listen for change events on (e.g., `SoftwareProposal`,
+   * `L10nConfig`).
+   */
+  scope: string;
+};
+
+/**
+ * Reactive alert shown when the configuration for a given scope has been
+ * changed externally.
+ *
+ * It warns that the interface may be out of sync and recommends reloading
+ * before continuing to avoid issues and data loss. Reloading is intentionally
+ * left up to the user rather than forced automatically, to prevent confusion
+ * caused by unexpected refreshes.
+ *
+ * It works by listening for "Changed" events on the specified scope:
+ *
+ * - Displays a toast alert if the event originates from a different client
+ *   (based on client ID).
+ * - Automatically dismisses the alert if a subsequent event originates from
+ *   the current client.
+ *
+ * @example
+ * ```tsx
+ * <AlertOutOfSync scope="SoftwareProposal" />
+ * ```
+ */
+export default function AlertOutOfSync({ scope, ...alertProps }: AlertOutOfSyncProps) {
+  const client = useInstallerClient();
+  const [active, setActive] = useState(false);
+  const missingScope = isEmpty(scope);
+
+  useEffect(() => {
+    if (missingScope) return;
+
+    return client.onEvent((event) => {
+      event.type === `${scope}Changed` && setActive(event.clientId !== client.id);
+    });
+  });
+
+  if (missingScope) {
+    console.error("AlertOutOfSync must receive a value for `scope` prop");
+    return;
+  }
+
+  const title = _("Configuration out of sync");
+
+  return (
+    <AlertGroup hasAnimations isToast isLiveRegion aria-live="assertive">
+      {active && (
+        <Alert
+          variant="info"
+          title={title}
+          actionClose={
+            <AlertActionCloseButton
+              title={title as string}
+              variantLabel={_("Out of sync alert")}
+              onClose={() => setActive(false)}
+            />
+          }
+          {...alertProps}
+          key={`${scope}-out-of-sync`}
+        >
+          <Content component="p">
+            {_(
+              "The configuration has been updated externally. \
+Reload the page to get the latest data and avoid issues or data loss.",
+            )}
+          </Content>
+          <Link to="." variant="link" isInline replace>
```

Comment:
Is this enough to trigger the kind of reload we need, for example, in the storage page?

I'm asking because usually clicking on "Storage" at the sidebar is not enough and you really need to trigger a browser reload (F5 or so) to see the changes.